### PR TITLE
feature: updated default token name/symbol

### PR DESCRIPTION
**Pull request by prestwich**
#459 

slightly refactors the way bridge tokens work.

Note that the name produced this way is 84 characters long.
The symbol is set to the first 32 characters of that.

I am not sure if long names and symbols will break anything

```
6 - optics
1 - . 
10 - domain
1 - .
2 - 0x
64 - identifier
---- 
= 84 characters long
```

### CODE REVIEW

**Comment by prestwich** [comment]
@anna-carroll this is ready for review and merge

File: solidity/optics-xapps/contracts/bridge/Encoding.sol [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+library Encoding {
+    bytes private constant NIBBLE_LOOKUP = "0123456789abcdef";
+
+    function encodeUint32(uint32 _num)
```
Comment (anna-carroll) [inline solidity/optics-xapps/contracts/bridge/Encoding.sol:7]: nit: this function is unused + undocumented - do we need it? if so, can we add docs?

File: solidity/optics-xapps/contracts/bridge/TokenRegistry.sol [hunk 2]
```diff
@@ -145,24 +146,47 @@ abstract contract TokenRegistry is Initializable, XAppConnectionClient {
 
     // ======== Internal Functions =========
 
-    function _cloneTokenContract() internal returns (address result) {
-        return address(new UpgradeBeaconProxy(tokenBeacon, ""));
+    function _defaultDetails(bytes29 _tokenId)
```
Comment (anna-carroll) [inline solidity/optics-xapps/contracts/bridge/TokenRegistry.sol:149]: TODO: add NatSpec docs to TokenRegistry internal functions (not blocking)

File: solidity/optics-xapps/contracts/bridge/TokenRegistry.sol [hunk 3]
```diff
@@ -145,24 +146,47 @@ abstract contract TokenRegistry is Initializable, XAppConnectionClient {
...
+        internal
+        pure
+        typeAssert(_tokenId, BridgeMessage.Types.TokenId)
+        returns (string memory _name, string memory _symbol)
+    {
+        // get the first and second half of the token ID
+        (uint256 _firstHalfId, uint256 _secondHalfId) = Encoding.encodeHex(uint256(_tokenId.id()));
```
Comment (anna-carroll) [inline solidity/optics-xapps/contracts/bridge/TokenRegistry.sol:156]: @prestwich can you walk me through the Encoding contract some time to explain how it works? this is out of my wheelhouse 🙂

**Review by anna-carroll — APPROVED** [review]
LGTM!! 🐋

**Pull request merged**